src/main/business/applicationFacade.js

```diff
@@ -99,63 +97,58 @@ class ApplicationFacade{
 
     /**
      * register a user
-     * @param pid - the process trying to sign up from
      * @param userdetails - details needed to register the user with
      * @returns {Promise<User>} - returns the created user
      * @throws Error - if the process is already logged in
      *               - if the username/email is taken
      */
-    async register(pid, userdetails){
+    async register(userdetails){
         // TODO: remove lines that add missing details
         if (!userdetails.firstName) userdetails.firstName = "Dummyfname";
         if (!userdetails.lastName) userdetails.lastName = "Dummylname";
         if (!userdetails.email) userdetails.email = `Dummyemail${this.#emailCounter++}@google.com`;
-        return (await this.userController.register(pid, userdetails));
+        return (await this.userController.register(userdetails));
     }
 
     /**
      * signs in user
-     * @param pid - the process trying to sign in
      * @param username - the user username - need to be registered
      * @param password - the user password
      * @returns {Promise<User>} - returned the signed-in user
      * @throws {Error} - if the user is already signed in
      *                 - if there is no registered user with this username
      *                 - if the password is incorrect
      */
-    async signIn(pid, username, password) {
-        return (await this.userController.signIn(pid, username, password));
+    async signIn(username, password) {
+        return (await this.userController.signIn(username, password));
     }
 
     /**
      * user wants to log out
-     * @param pid - the process log out
      * @throws {Error} - if the user is not signed in
      */
-    logout(pid) {
-        return this.sessionManager.logout(pid)
+    //TODO:REMOVE
+    logout(logoutData) {
+        //Log out should be done in client side, by removing the cookies.
     }
 
     /**
      * change password
-     * @param pid - a logged-in session
      * @param newPassword - updated password for the logged user
      */
-    async changePassword(pid, newPassword) {
-        const user = await this.sessionManager.getUser(pid);
-        await user.changePassword(newPassword);
+    async changePassword(changePasswordData) {
+        const user = await this.userController.getUser(changePasswordData.callingUser.username);
+        await user.changePassword(changePasswordData.newPassword);
```

Comment:
sometimes you pass filtered information like in here, sometimes you pass all the information like in `getAllUsers` 
what will it be?

Reply:
I just adjust it to the previous code, I look at the signature and the function and looked what it expect to receive.

I wont do a full refuctor of the code in this branch

Reply:
You leave the code less readable then it was...

Reply:
I going to do a big refuctor to the code, that every function will have verify at the beggining of its function, validate all require fields are present.

That will solve the issue and will act as signature.

Reply:
But it will be in a different branch